### ISSUE TITLE
Update 0.2.4 -> 0.2.6 causes build errors on Heroku

### ISSUE THREAD

**Issue by oskargargas** [logtail/logtail-ruby-rails#37]
Hi!

After updating logtail-rails to 0.2.6 I started experiencing build errors on Heroku (locally everything seams to work fine).
What might be misleading is that error is not directly related to logtails. Rather to Sentry gem which also uses `ActiveSupport::BroadcastLogger` and probably gets into some kind of loading priority problems.

```
Preparing app for Rails asset pipeline
Running: rake assets:precompile
rake aborted!
NameError: uninitialized constant ActiveSupport::BroadcastLogger (NameError)

         @logger = ::ActiveSupport::BroadcastLogger.new(*dupped_broadcasts)
                                  ^^^^^^^^^^^^^^^^^
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-rails-5.12.0/lib/sentry/rails/configuration.rb:17:in `block in <class:Configuration>'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry/configuration.rb:608:in `instance_eval'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry/configuration.rb:608:in `block in run_post_initialization_callbacks'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry/configuration.rb:607:in `each'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry/configuration.rb:607:in `run_post_initialization_callbacks'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry/configuration.rb:353:in `initialize'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry-ruby.rb:210:in `new'
/tmp/build_30e21af1/vendor/bundle/ruby/3.2.0/gems/sentry-ruby-5.12.0/lib/sentry-ruby.rb:210:in `init'
/tmp/build_30e21af1/config/initializers/sentry.rb:4:in `<main>'
```

`logtail-rails` was the only updated dependency at this point.

**Comment by oskargargas** [comment]
Actually narrowed it to 0.2.4 -> 0.2.5

**Comment by curusarn** [comment]
Hi @oskargargas,

Thank you for reporting this!

I've passed the issue to the team to take a look. I'll make sure to keep you updated here on GitHub.


Thanks for sharing logs and versions!

**Comment by PetrHeinz** [comment]
Hello @oskargargas, and thanks again for reporting the issue! 🙌 

However, the error triggers because `sentry-ruby` package doesn't check whether `BroadcastLogger` is defined before instantiating it in their code. I've proposed a fix in their code in https://github.com/getsentry/sentry-ruby/pull/2169, but merging and releasing depends on their maintainers.

In the meantime, I would suggest either monkey-patching `sentry-ruby` (in a similar way as the PR suggests), upgrading your Rails to 7.1+, or using `logtail-rails` in v0.2.4 for the time being.

---

Since there's not much we can do about it from our side, I'm closing the issue.

Feel free to reopen it, if you have any more questions or an idea how to work around the issue from `logtail-rails`. 🙏

**Comment by PetrHeinz** [comment]
Hello @oskargargas, my PR was merged into `sentry-ruby` and will be probably released in `5.14.0` in a matter of weeks.

Updating to it when it's released should resolve the issue you're having. 👍

**Comment by oskargargas** [comment]
@PetrHeinz thanks for your help and information provided. Greatly appreciated!
